### PR TITLE
Encode float operators with a dot

### PR DESCRIPTION
**Pull request by nilehmann**
I noticed some of the float operators were missing while working on [this](https://github.com/xldenis/creusot/discussions/735). I got the syntax for operators from here https://why3.lri.fr/stdlib/ieee_float.html. 

I added a test, but not sure if that's how it should be tested. I tried it locally and it worked.

### CODE REVIEW

File: creusot/src/backend/term.rs [hunk 1]
```diff
@@ -392,7 +392,10 @@ pub(crate) fn lower_literal<'tcx>(
             names.insert(id, subst);
             Exp::Tuple(Vec::new())
         }
-        Literal::Float(f) => Constant::Float(f).into(),
+        Literal::Float(f, fty) => {
+            let _why_ty = floatty_to_ty(names, &fty);
```
Comment (nilehmann) [inline creusot/src/backend/term.rs:396]: I needed this to import the module. Not sure if should be propagated down in the `Constant::Float`.

File: why3/src/exp.rs [hunk 2]
```diff
@@ -50,11 +54,15 @@ impl BinOp {
             BinOp::Ne => Infix1,
             BinOp::Ge => Infix1,
             BinOp::Gt => Infix1,
-            BinOp::FloatAdd => Infix4,
-            BinOp::FloatSub => Infix4,
-            BinOp::FloatMul => Infix4,
-            BinOp::FloatDiv => Infix4,
+            BinOp::FloatAdd => Infix5,
```
Comment (nilehmann) [inline why3/src/exp.rs:57]: I added a new precedence level to separate arithmetic from comparison for float operators. That felt sensible to me.

Comment (xldenis) [reply why3/src/exp.rs:57]: Hmm the why manual lists the precedence levels of operators and how they should spread out throughout them. According to that all float operators should be Infix4

**Comment by xldenis** [comment]
Hi, thank you for your contribution to Creusot! I just have one minor comment about precedence of the new operators but otherwise this looks good to me.

**Comment by nilehmann** [comment]
Oh, I thought they had two levels. Removed it!

**Pull request merged**